client/src/test/java/com/sinch/sdk/domains/sms/adapters/BatchesServiceTest.java

```diff
@@ -376,10 +379,18 @@ void list() throws ApiException {
             eq(null),
             eq(null)))
         .thenReturn(listBatchesResponseDtoPage1);
-
+    when(api.listBatches(
+            eq(configuration.getProjectId()),
+            eq(2),
+            eq(null),
+            eq(null),
+            eq(null),
+            eq(null),
+            eq(null)))
+        .thenReturn(listBatchesResponseDtoPage2);
     BatchesListResponse response = service.list(null);
 
-    Iterator<Batch<?>> iterator = response.autoPageIter();
+    Iterator<Batch<?>> iterator = response.iterator();
```

Comment:
BTW worth checking if there is no additional items in the end of the iterator

Reply:
Yes, this what is done at line 433: `Assertions.assertThat(iterator.hasNext()).isEqualTo(false);`